src/bench/bench.cpp

```diff
@@ -94,9 +100,13 @@ bool benchmark::State::KeepRunning()
     assert(count != 0 && "count == 0 => (now == 0 && beginTime == 0) => return above");
 
     // Output results
-    double average = (now-beginTime)/count;
+    // Duration casts are only necessary here because hardware with sub-nanosecond clocks
+    // will lose precision.
+    int64_t min_elapsed = std::chrono::duration_cast<std::chrono::nanoseconds>(minTime).count();
+    int64_t max_elapsed = std::chrono::duration_cast<std::chrono::nanoseconds>(maxTime).count();
+    int64_t avg_elapsed = std::chrono::duration_cast<std::chrono::nanoseconds>((now-beginTime)/count).count();
     int64_t averageCycles = (nowCycles-beginCycles)/count;
-    std::cout << std::fixed << std::setprecision(15) << name << "," << count << "," << minTime << "," << maxTime << "," << average << ","
+    std::cout << std::fixed << std::setprecision(15) << name << "," << count << "," << min_elapsed << "ns," << max_elapsed << "ns," << avg_elapsed << "ns,"
```

Comment:
Please put the unit id in the row header, not in the data. Adding it to every record complicates parsing (for example, spreadsheets won't see it as number anymore, and will not be able to compute with it or plot it).